components/datetime/src/format/datetime.rs

```diff
@@ -299,8 +320,15 @@ where
                     let symbols = date_symbols
                         .ok_or(Error::MissingDateSymbols)?
                         .get_symbols_for_month(month, length)?;
-                    w.write_str(symbols.get(code).ok_or(Error::MissingMonthSymbol(code))?)?;
-                    w.write_str("(leap)")?; // This is temporary; TODO(#3766) add support for leap months
+                    let symbol = symbols.get(code).ok_or(Error::MissingMonthSymbol(code))?;
+                    // FIXME (#3766) this should be using actual data for leap months
+                    let leap_str = match datetime.any_calendar_kind() {
+                        Some(AnyCalendarKind::Chinese) => CHINESE_LEAP_PREFIX,
+                        Some(AnyCalendarKind::Dangi) => DANGI_LEAP_PREFIX,
+                        _ => "(leap)",
```

Comment:
leap months are not a chinese concept. cyclic years are (the dangi ones are derived from the chinese ones and it would actually be 100% valid to use the chinese ones there)